### ISSUE TITLE
Order of argument definitions (variable, output, module)

### ISSUE THREAD

**Issue by nitrocode** [terraform-linters/tflint#1384]
It would be nice if we can create a linting rule to enforce the order of arguments for vars, outputs, and modules

<details><summary>terraform_module_argument_order</summary>

```hcl
rule "terraform_module_argument_order" {
  enabled = true
  order   = ["source", "version"]
}
```

This would work

```hcl
module "example" {
  source  = ""
  version = ""
}
```

This would fail

```hcl
module "example" {
  version = ""
  source  = ""
}
```

</details>

<details><summary>terraform_variable_argument_order</summary>

```hcl
rule "terraform_variable_argument_order" {
  enabled = true
  order   = ["type", "description", "default", "sensitive", "validation"]
}
```

This would work

```hcl
variable "example" {
  type        = "string"
  description = ""
  default     = ""
  sensitive   = true
  validation {}
}
```

This would fail

```hcl
variable "example" {
  description = ""
  type        = "string"
  default     = ""
  sensitive   = true
  validation {}
}
```

</details>

<details><summary>terraform_output_argument_order</summary>

```hcl
rule "terraform_output_argument_order" {
  enabled = true
  order   = ["description", "value"]
}
```

This would work

```hcl
output "example" {
  description = ""
  type        = ""
}
```

This would fail

```hcl
output "example" {
  type        = ""
  description = ""
}
```

</details>

**Comment by nitrocode** [comment]
cc: @PatMyron could you add some comments?

**Comment by lonegunmanb** [comment]
Hi @wata727 , we've tried to implement such rule for `resource` and `data` block, but arguments' order in those two blocks depends on argument's `required`/`optional` property, which can only be fetched via provider's source code. We'd like to implement such rule in our own plugin so we can reference provider's source code there, but we found that `Runner` type in plugin sdk is different than `Runner` in tflint core. Is there any way for us to leverage powerful `Runner.TFConfig` information in plugin project to save effort? Many thanks!

**Comment by wata727** [comment]
> arguments' order in those two blocks depends on argument's required/optional property,

You can see their order from `hcl.Range`. Is this not enough?
